source/dub/commandline.d

```diff
@@ -836,12 +837,54 @@ class BuildCommand : GenerateCommand {
 		args.getopt("f|force", &m_force, [
 			"Forces a recompilation even if the target is up to date"
 		]);
+		args.getopt("y|yes", &m_yes, [
+			`Automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.`
+		]);
 		super.prepare(args);
 		m_generator = "build";
 	}
 
 	override int execute(Dub dub, string[] free_args, string[] app_args)
 	{
+		if (free_args.length >= 1) {
+			auto package_name = free_args[0];
+			auto pack = dub.packageManager.getFirstPackage(package_name);
+
+			if (!pack) {
+				bool input(string caption, bool default_value = true) {
+					writef("%s [%s]: ", caption, default_value ? "Y/n" : "y/N");
+					auto inp = readln();
+					string userInput = "y";
+					if (inp.length > 1)
+						userInput = inp[0 .. $ - 1].toLower;
+
+					switch (userInput) {
+						case "no", "n", "0":
+							return false;
+						case "yes", "y", "1":
+						default:
+							return true;
+					}
+				}
+
+				// search for the package and filter for exact matches
+				auto search = dub.searchPackages(package_name)
+					.map!(tup => tup[1].find!(p => p.name == package_name))
+					.filter!(ps => !ps.empty);
+				if (search.empty)
+					return 2;
+
+				auto p = search.front.front;
+				writefln("%s wasn't found locally, but it's available online:", package_name);
+				writefln("Description: %s", p.description);
+				writefln("Version: %s", p.version_);
+
+				bool answer = m_yes ? true : input("Do you want to fetch %s?".format(package_name));
```

Comment:
I like the change but I think we should move the m_yes/input method into a more general scope where all methods could read a boolean input

Reply:
Is there anything else that currently uses this?
We could have easily done this after this PR has been merged and this simple improvement has been in the queue now for more than 1 1/2 years (see the original PR).